### PR TITLE
Add `key public` cli command

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#1389 

# Overview

- [x] Quick implementation of `key public`, with a golden test

# Comments

### CODE REVIEW

**Review by paweljakubas — APPROVED** [review]
looks ok modulo missing part in CLISpec (something like below): 
```
        ["key", "public", "--help"] `shouldShowUsage`
            [ "Usage:  key inspect HEX-XPRV"
            , "  Extract the public key from a given private key.."
            , ""
            , "Available options:"
            , "  -h,--help                Show this help text"
            ]
```

**Comment by Anviking** [comment]
Added usage test 👍 

I cherry-picked 1815b96b26b3e92682953ee6a44070d08070684e (⚠️, but it is tiny) from #1370, to be able to merge this and #1397 before #1370.

bors r+

File: lib/cli/test/unit/Cardano/CLISpec.hs [hunk 1]
```diff
@@ -466,6 +466,13 @@ spec = do
             , "                             trezor (12, 15, 18, 21 or 24 words)"
             , "                             ledger (12, 15, 18, 21 or 24 words)"
             ]
+        ["key", "public", "--help"] `shouldShowUsage`
+            [ "Usage:  key public HEX-XPRV"
```
Comment (KtorZ) [inline lib/cli/test/unit/Cardano/CLISpec.hs:470]: *XPRV

The meta-type shouldn't mix the purpose with the encoding. (The meta-type says _what_ is expected, not _how_).

Comment (Anviking) [reply lib/cli/test/unit/Cardano/CLISpec.hs:470]: Fixing in #1370

**Pull request merged**